server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts

```diff
@@ -246,82 +249,191 @@ export default class ProbationPractitionerReferralsController {
+
+  async backwardsCompatibilityUpdateCancellationReason(req: Request, res: Response): Promise<void> {
     const { user } = res.locals
     const { accessToken } = user.token
+    const referralId = req.params.id
 
-    const sentReferral = await this.interventionsService.getSentReferral(accessToken, req.params.id)
+    const data = await new ReferralCancellationReasonForm(req).data()
+
+    let formError: FormValidationError | null = null
+
+    if (req.method === 'POST') {
+      if (!data.error) {
+        const draftCancellation = await this.draftsService.createDraft<DraftCancellationData>(
+          'cancellation',
+          data.paramsForUpdate,
+          { userId: res.locals.user.userId }
+        )
+
+        return res.redirect(
+          `/probation-practitioner/referrals/${referralId}/cancellation/${draftCancellation.id}/check-your-answers`
+        )
+      }
+
+      res.status(400)
+      formError = data.error
+    }
+
+    const sentReferral = await this.interventionsService.getSentReferral(accessToken, referralId)
     const intervention = await this.interventionsService.getIntervention(
       accessToken,
       sentReferral.referral.interventionId
     )
     const serviceUser = await this.communityApiService.getServiceUserByCRN(sentReferral.referral.serviceUser.crn)
     const cancellationReasons = await this.interventionsService.getReferralCancellationReasons(accessToken)
 
+    // We don’t use this draft, other than to pass it to the presenter for rendering
+    // (remember this controller action is only temporary)
+    const draftCancellation = await this.draftsService.createDraft<DraftCancellationData>(
+      'cancellation',
+      {
+        cancellationReason: null,
+        cancellationComments: null,
+      },
+      { userId: res.locals.user.userId }
+    )
+
     const presenter = new ReferralCancellationReasonPresenter(
+      draftCancellation,
       sentReferral,
       intervention,
       serviceUser,
-      cancellationReasons
+      cancellationReasons,
+      formError
     )
+
+    await this.draftsService.deleteDraft(draftCancellation.id, { userId: res.locals.user.userId })
+
     const view = new ReferralCancellationReasonView(presenter)
 
     return ControllerUtils.renderWithLayout(res, view, serviceUser)
   }
 
-  async submitFormAndShowCancellationCheckAnswersPage(req: Request, res: Response): Promise<void> {
+  private async fetchDraftCancellationOrThrowSpecificError(req: Request, res: Response) {
+    const id = req.params.draftCancellationId
+    const draftCancellation = await this.draftsService.fetchDraft<DraftCancellationData>(id, {
+      userId: res.locals.user.userId,
+    })
+
+    if (draftCancellation === null) {
+      throw createError(500, `Draft cancellation with ID ${id} not found by drafts service`, {
+        userMessage:
+          'Too much time has passed since you started cancelling this referral. Your answers have not been saved, and you will need to start again.',
```

Comment:
nit: I wonder if it would be nicer to indicate to the user what "too much time" means. Maybe something like:

"Unable to retrieve draft as it has been over 24 hours since it has been accessed."

(maybe asking UX design on best wording)

Reply:
Yeah, I think it's unlikely they'll see it, as you say. I'd prefer not to expose the concept of a "draft" to the user since it's kinda just an implementation detail that we don't mention elsewhere in the journey. But perhaps I should be explicit about the time period (problem is that it's configurable so might be a slight faff to get the text right, but will take a look).

Reply:
(Also, if we see that this error is actually happening, then we can iterate and make this page friendlier. Or just bump the timeout.)

Reply:
Yeah, looking at it I think it's going to be a faff to do this with the configurable timeout (we have to do it in two places too). Reckon we can leave it until we find out whether it actually happens in real life?